wordpress-mobile/WordPress-Android
# Merging reblog feature branch into develop.

Fixes #10934 

This PR merges the reblog feature branch into develop.

Taken from the linked issue:

> Let's allow a user to reblog a post that they're viewing in the Reader.
We'll add a reblog button to post cards in a stream view, as well as to the actions in a detail view (along with commenting, liking, saving, etc). When tapped we'll open the editor, prepopulated with the shared post's title, snippet, and featured image if there is one.

We have added a reblog button in post detail screen and in reader post list streams (unfiltered or filtered)

| ![image](https://user-images.githubusercontent.com/47797566/80807564-d6614480-8bbd-11ea-938c-190b049167ed.png) | ![image](https://user-images.githubusercontent.com/47797566/80807610-f8f35d80-8bbd-11ea-9601-e08bfa49e168.png) |
|---|---|

The reblog operation ends up in the editor pre-populated from where user should be able to continue with usual operation like save as draft/publish.

## Reblog button availability
The reblog button is available only to users that are logged into WP.com and on sites (posts) that are not private.
A user can reblog a post only to a WP.com/JP/Atomic site that is marked as `visible` in the main site picker (not possible to reblog to a self-hosted site even if visible in main site picker)

## Reblog 3 cases

1. User with only 1 visible WP.com/JP/Atomic site: the reblog action lands directly to the editor pre-populated with reblog post content

<p align="center">
  <img width="300" src=https://user-images.githubusercontent.com/47797566/80807767-4d96d880-8bbe-11ea-96d7-6411ab7995a8.png>
</p>

2. User with more than 1 visible WP.com/JP/Atomic site: the reblog action opens the site picker allowing to select the site to which to reblog (only visible WP.com/JP/Atomic site are listed); after site selection the editor is opened

| ![image](https://user-images.githubusercontent.com/47797566/80807821-6e5f2e00-8bbe-11ea-829d-445aa92ddd88.png) | ![image](https://user-images.githubusercontent.com/47797566/80808013-e0377780-8bbe-11ea-94f0-d279b2b44a38.png) |
|---|---|

3. User with no visible WP.com/JP/Atomic site: the reblog action opens the `no site` screen, tapping on `Manage Sites` user lands to `My Site`

<p align="center">
  <img width="300" src=https://user-images.githubusercontent.com/47797566/80808061-fcd3af80-8bbe-11ea-8b3b-a0b662082ca3.png>
</p>


## Main Linked PRs

- Add reblog to Reader Post Details #11534 
- Add reblog to reader post list #11540 
- Handle user with no site #11553 
- Reblog feature: small customization of site picker #11709 
- Adding reblog dedicated tracking events #11788 
- Reblog to wpcom sites only #11795 

## To test
- Reblog feature must work both with Aztec and GB
- Follow a private site and check reblog button is not available on posts from that private site
- Smoke test the reblog functionality and the other parts of the app (try the reblog both for posts with/without featured image; reblog to WP.com/JP/Atomic sites)
- Further test steps in the PRs linked above

PR submission checklist:
